Document all command line options

per1234 opened the pull request:
This merges all the content from https://github.com/EventGhost/EventGhost/wiki/Command-Line-Options

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -3,18 +3,45 @@ Command Line Options
+    the application data folder of your machine for storing its settings.
+    For example, through this option you can change the folder to a location
+    on a USB stick to make EventGhost portable.
+
```
I would remove the comments about being portable. because EG technically does not conform to being a portable application because of its creation/use of ProgramData\EventGhost. and it does not remove the folder after being run. so it leaves behind data. and a portable program will not/should not do this.

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -3,18 +3,45 @@ Command Line Options
     added to the event in the 
     :data:`eg.event.payload <eg.EventGhostEvent.payload>` field.
 
+
```
there is one other thing you should mention here. if EG was run with elevated privileges in order for the trigger event to work from a command line that has to be run with elevated privileges as well.

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -3,18 +3,45 @@ Command Line Options
+
+    Opens the XML configuration file <filename> on start-up (instead of the
+    last loaded file).
+
```
Opens .egtree user save data

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -3,18 +3,45 @@ Command Line Options
+
 .. cmdoption:: -hide
 
     Starts EventGhost hidden in the system tray. Otherwise it would start in
```
if you have EG set to display a tray icon it will open EG minimized to the tray. if not it will open EG minimized to the task bar

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -26,37 +53,56 @@ The EventGhost main executable accepts the following command line arguments:
 
-    Opens the XML configuration file <filename> on start-up (instead of the 
-    last loaded file). 
+    Allow multiple sessions of EventGhost on the same computer.
```
If the first session of EG has been run with elevated privileges all subsequent sessions bust have elevated permissions as well

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -26,37 +53,56 @@ The EventGhost main executable accepts the following command line arguments:
+    other applications or .BAT files to send events to a remote machine.
+    <host> has to be the IP or host name of the target machine. <port> and
+    <password> are the options that you have configured on the target
+    machine's 'Network Event Receiver' plugin.
```
If EG has been run with elevated privileges this has to be run elevated as well

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -26,37 +53,56 @@ The EventGhost main executable accepts the following command line arguments:
+    Opens the plugin archive file <egplugin> for installation in EventGhost.
+    Plugins installed through a <egplugin> plugin archive file are stored in the
+    %AppData%\EventGhost\plugins folder and take precedence over plugins in the
+    %ProgramFiles%\EventGhost\plugins folder.
```
I would have to check on this. But i do know there is an issue that was brought up and I made a change on and I have yet to hear back if it works properly. But it has to do with EG running elevated and then double clicking the .egplugin file and an error being created because you have to run the .egplugin file elevated as well.

kdschlosser commented on _build/data/docs/commandline.rst:
```diff
@@ -26,37 +53,56 @@ The EventGhost main executable accepts the following command line arguments:
+
+    Restart EventGhost.
+
+
```
If EG has been run elevated this command has to be run elevated as well.

per1234 commented:
@kdschlosser thanks, great feedback! I have made some changes to fix everything except for the elevated privileges issues. Do you think it would make more sense to just add a note at the top of the document:

>Note that if you have started EventGhost with elevated privileges (including **File > Restart as Administrator**) you must also run command line options sent to EventGhost with the same privileges.

kdschlosser commented:
Well that would be a really good question. I know how I personally look through documentation. I jump right to the piece I am looking for and read about it. I would fly right by a single line comment about it that wasn't with that specific command. But that's me. I can see the purpose to not repeating as well. so maybe the best of both worlds. but a marker and add a comment line at the bottom maybe. I do read those if i see them. whatever you would like to do.

Also one other thing. I forgot to post a comment on it. But i don't think i saw anything for the -d shortcut for -debug. and one other thing is if you simply do

    eventghost.exe savefile.egtree

or

    eventghost.exe plugin.egplugin

it will sort out what it needs to and either open that save file or install that plugin

per1234 commented:
> I jump right to the piece I am looking for and read about it. I would fly right by a single line comment about it that wasn't with that specific command.

OK, I went ahead and added a duplicate note to each option you mentioned in your review.

> i don't think i saw anything for the -d shortcut for -debug

Done

>it will sort out what it needs to and either open that save file or install that plugin

Done

kdschlosser commented:
oh crazy thing. I am sorry for this @per1234 I am going to be able to solve the admin issues 👍 but the work you just did will not be necessary. I have found a way to get EG to run as elevated if an error pops up.
i didn't think it would happen this fast the idea kinda just popped into my head and when i did a little research i discovered that my idea will indeed work. I have to code it up and test it out. but i feel really confident that it will fix the issue. tho i have been confident and failed before. so don't remove the changes as of yet.

per1234 commented:
Alternately, we could merge this as is now since this is correct documentation for the current code, then you could remove the documentation notes about elevated privileges in the same commit that you fix the problem. It just depends on which order the PRs get merged in. This one seems like it could be merged more quickly since it's ready now and doesn't contain any code changes that need testing.

per1234 commented:
@kdschlosser, am I correct to think that now that https://github.com/EventGhost/EventGhost/pull/240 has been merged I should remove all the notes about elevated privileges from this PR?

topic2k commented:
Crazy 😆 i was just about to comment :D

remove the notes, except for the restart option: If EG was previously started with full elevation and you run EG -restart from a commandline with limited elevation, EG will be restarted with limited elevation. So the user has to take care of this.

topic2k commented:
Use the following form to highlight the note:

```rst
    .. note::

        If you started EventGhost with full elevation you must also run
        this commands with full elevation.
```

per1234 commented:
OK, I removed all the notes about elevation except for `-restart`, changed the format of the note on `-restart` as requested, and did a squash + force push.

topic2k commented:
Thank you. Just as an info: No need to squash the commits. Thats make it easier to see what changed while the PR evolved. I can then squash-merge the PR.

per1234 commented:
OK, I will do that in the future. I do the squash+force push because a lot of times people merging PRs don't bother and I'm uptight about clean commit histories.

per1234 commented:
No worries. It's actually an argument against me squashing fix up commits in a PR because if had added those notes in a separate commit I could have just reverted it. But it wasn't a big deal either way. It is still important to note the elevated privileges thing for `-restart` so it's good you made that suggestion. I'm just happy that the docs now have all the command line options

kdschlosser commented:
well You are going to end up at some point removing that not for the -restart as well. I am currently fixing the endless loop issue that could happen if EG has unsaved data.
